### DISCUSSION TITLE
Features

### DISCUSSION THREAD

**Discussion by GetStreamlined** [delucca/chia-seeder#16] [Ideas]
Hey Delucca

Thanks for your help the other day. Just wanted to stop by and give a brief list of some features that I personally would love to see in a plotter.

- No binaries - having executable binaries makes code non-transparent and risky - you should keep it open source if possible!
- Ability to choose between chia plotter and madmax
- Longshot... ability to use a 50GB ramdisk to help speed it up and reduce TBW on plotting drives
- Ability to separate phase 5 from the plotter. Most plotters do not kick off another plot until the copy part is done - I would like to!
- Log output - logs should/could be in the same format as chia-plotter so that other analysis tools would not need to retro fit your logs in.
- Metrics - it would be nice to see the current HDD speeds (and CPU) tracked in the plotter. Nice to have but i know task managers like windows dont report this correctly.
- Windows - Please ensure that all tasks are kicked off in their own process. I think using swar with madmax currently puts it as a system process in windows (and therefore not killable) - would be nice to have the ability to kill the process. (Maybe this can be a command line option in the plotter?)

Anyway... focus on your masters and good luck!

James

**Comment by delucca** [comment]
Hey @GetStreamlined thank you for your post here :) It is amazing for me to hear out from possible future users regarding their desires! That helps me a lot to coding the right features!

I'm going to comment on some of the features you suggested:

> No binaries - having executable binaries makes code non-transparent and risky - you should keep it open source if possible!
100% agree! That is my idea. I love the open-source community and I would never do such a thing 😄 

> Ability to choose between chia plotter and madmax
👍🏻 100% agree! That is one of the main ideas! I'm going to add it to the README to make it be pretty clear for future users

> Longshot... ability to use a 50GB ramdisk to help speed it up and reduce TBW on plotting drives
That is really, really interesting! I'm going to note it tonight in my backlog!

> Ability to separate phase 5 from the plotter. Most plotters do not kick off another plot until the copy part is done - I would like to!
👍🏻 

> Log output - logs should/could be in the same format as chia-plotter so that other analysis tools would not need to retro fit your logs in
👍🏻 

> Metrics - it would be nice to see the current HDD speeds (and CPU) tracked in the plotter. Nice to have but i know task managers like windows dont report this correctly.
Nice idea! I drawed an architecture for an `overseer` component that would manage statistics. That is a pretty good use-case for it!

> Windows - Please ensure that all tasks are kicked off in their own process. I think using swar with madmax currently puts it as a system process in windows (and therefore not killable) - would be nice to have the ability to kill the process. (Maybe this can be a command line option in the plotter?)
👍🏻 

Thanks!!! Cheers (:

**Comment by GetStreamlined** [comment]
Thanks for the responses. On the 50GB Ram Disk idea, i was saying that from a purely selfish perspective. My system has 64GB ram and i use up to 14gb on general use, chrome etc - hell i can even use the page file if i needed to. I've seen many suggestions about using ram drives but everyone is talking 128GB/256GB for it to be worth it but surely there has to be some sort of trade off here that would potentially utilise smaller chunks and take slightly longer whilst preserving the hard disks. Surely the baskets had this kind of thing in mind?

James

**Reply by delucca** [reply]
> Thanks for the responses. On the 50GB Ram Disk idea, i was saying that from a purely selfish perspective. My system has 64GB ram and i use up to 14gb on general use, chrome etc - hell i can even use the page file if i needed to. I've seen many suggestions about using ram drives but everyone is talking 128GB/256GB for it to be worth it but surely there has to be some sort of trade off here that would potentially utilise smaller chunks and take slightly longer whilst preserving the hard disks. Surely the baskets had this kind of thing in mind?
> 
> James

Oh! I understand now!

Unfortunately, that is not related to the plot manager itself. Since the plot can't be stopped/resumed (yet) you need at least 128GB to run a ramdisk on tmpfs2.

It isn't possible to achieve anything below than that, at least in a plot manager 😞

**Comment by GetStreamlined** [comment]
It was a longshot at best - thought i'd ask anyway ;)

James
